src/debugger.ts

```diff
@@ -2,32 +2,70 @@ import * as vscode from 'vscode';
 import * as path from 'node:path';
 import { globalContext } from './extension';
 import { MirrordConfigManager } from './config';
-import { LAST_TARGET_KEY, MirrordAPI, mirrordFailure } from './api';
+import { LAST_TARGET_KEY, MirrordAPI, mirrordFailure, MirrordExecution } from './api';
 import { updateTelemetries } from './versionCheck';
 import { getLocalMirrordBinary, getMirrordBinary } from './binaryManager';
+import { platform } from 'node:os';
 
-/// Get the name of the field that holds the exectuable in a debug configuration of the given type.
-function getExecutableFieldName(config: vscode.DebugConfiguration): keyof vscode.DebugConfiguration {
+const DYLD_ENV_VAR_NAME = "DYLD_INSERT_LIBRARIES";
+
+/// Get the name of the field that holds the exectuable in a debug configuration of the given type,
+/// and the executable. Returning the field name for replacing the value with the patched path later.
+/// Also returning the executable because in some configuration types there is some extra logic to
+/// be done for retrieving the executable out of its field (see the `node-terminal` case).
+function getFieldAndExecutable(config: vscode.DebugConfiguration): [keyof vscode.DebugConfiguration, string | null] {
 	switch (config.type) {
 		case "pwa-node":
 		case "node": {
-			return "runtimeExecutable";
+			return ["runtimeExecutable", config["runtimeExecutable"]];
+		}
+		case "node-terminal": {
+			return ["command", config["command"]?.split(' ')[0]];
 		}
 		case "python": {
 			if ("python" in config) {
-				return "python";
+				return ["python", config["python"]];
 			}
 			// Official documentation states the relevant field name is "python" (https://code.visualstudio.com/docs/python/debugging#_python), 
 			// but when debugging we see the field is called "pythonPath".
-			return "pythonPath";
+			return ["pythonPath", config["python"]];
 		}
 		default: {
-			return "program";
+			return ["program", config["program"]];
 		}
 	}
+}
 
+/// Edit the launch configuration in order to sidestep SIP on macOS, and allow the layer to be
+/// loaded into the process. This includes replacing the executable with the path to a patched
+/// executable if the original executable is SIP protected, and some other special workarounds.
+function changeConfigForSip(config: vscode.DebugConfiguration, executableFieldName: string, executionInfo: MirrordExecution) {
+	if (config.type === "node-terminal") {
+		let command = config[executableFieldName];
+		if (command === null) {
+			return;
+		}
+		if (executionInfo.patchedPath !== null) {
+			// replace the first word of the command line with a patched version of the executable.
+			let words = command.split(' ');
+			words[0] = executionInfo.patchedPath;
```

Comment:
`command && executionInfo.patchedPath` perhaps?

Reply:
That would change the meaning though - if command is null - we also don't want to do the stuff that's after this if.

Reply:
you return when command is null, instead if you just do:

```
if (command && patchedPath) {
      const words = command.split(' ');
      words[0] = patchedPath;
      command = words.join(' ');
      const libraryPath = executionInfo.env.get(DYLD_ENV_VAR_NAME);
      config[executableFieldName] = `${DYLD_ENV_VAR_NAME}=${libraryPath} ${command}`;
    }
 ```
 wouldnt it mean the same?

Reply:
No - the last two lines inside the `if` in your snippet are only executed if `patchedPath` is not null, but they should be executed regardless of that.

Reply:
alrighty